src/Spago/Git.purs

```diff
@@ -40,14 +41,19 @@ runGit args cwd = ExceptT do
     Right r -> Right r.stdout
     Left r -> Left r.stderr
 
-fetchRepo :: forall a b. { git :: String, ref :: String | a } -> FilePath -> Spago (GitEnv b) (Either (Array String) Unit)
+determineShouldCache :: forall b. String -> FilePath -> Spago (GitEnv b) Boolean
+determineShouldCache ref path = do
```

Comment:
It's fine to name things after their purpose, but I'd lean towards being more general here, for the sake of the casual reader (or me in a few months) which might struggle to see how a random git command is tied to our cache. I'd also link the StackOverflow answer for more context:

```suggestion
-- See https://stackoverflow.com/questions/18222634
isBranch :: forall b. { path :: FilePath, ref :: String } -> Spago (GitEnv b) Boolean
isBranch { ref, path } = do
```